Restructuring Sygma docs

haochizzle opened the pull request:
Restructure Sygma docs based on #19. 

- `Introduction` rename to `What Is Sygma`
- Merge `Abstract` with `Origins`
- Move `Sygma SDK` section above `Architecture`
- Update `Audits` to include a table with HAECHI and Least Authority audits 
- Add `draft: true` subpages to:
  - `What Is Sygma`
    - `Vision`
  - `Sygma SDK`
    - `Quick Start` section and all subpages
    - `Advanced` section and all subpages
    - `Examples` section and all subpages

This branch will continue to be iterated on as we fill these out.

MakMuftic commented:
@haochizzle I like the new structure :fire: 
Here are some suggestions:

- I think we should remove the "Using The SDK" page since it's an outdated README from the SDK. It doesn't seem relevant anymore, especially since we have the Quick-Start guide that should cover the basics of using the SDK.
- We need to decide on a consistent term to refer to GMP or xCall or some other term. It would be best to settle on one term and use it consistently across all our documents. cc: @alexmmueller 

Additionally you will see that I changed based branch to `v2`, I think it is better to prepare some of these new pages and collect them in one branch. Once we're satisfied with the progress, we can merge the `v2` branch and deploy the new version of Sygma docs.

MakMuftic commented:
@haochizzle we decided that for now we are going to stick with the general message passing (GMP) naming.

MakMuftic commented:
> @haochizzle we decided that for now we are going to stick with the general message passing (GMP) naming.

@waymobetta sorry forgot to tag you here

waymobetta commented on docs/02-sygma-sdk/03-Quick-Start/03-xcall-gmp.md:
we'll want to replace all instances of `xcall` in favor of `GMP` to standardize; change filename as well and remove from dictionary

haochizzle replied:
done!

waymobetta reviewed (changes requested):
Looks good just remove instances of `xcall`

haochizzle commented:
> @haochizzle I like the new structure 🔥 Here are some suggestions:
> 
> * I think we should remove the "Using The SDK" page since it's an outdated README from the SDK. It doesn't seem relevant anymore, especially since we have the Quick-Start guide that should cover the basics of using the SDK.
> * We need to decide on a consistent term to refer to GMP or xCall or some other term. It would be best to settle on one term and use it consistently across all our documents. cc: @alexmmueller
> 
> Additionally you will see that I changed based branch to `v2`, I think it is better to prepare some of these new pages and collect them in one branch. Once we're satisfied with the progress, we can merge the `v2` branch and deploy the new version of Sygma docs.

Addressing point 1: wasn't sure if we should keep it because the SDK refactor is not completed yet. So the "Using The SDK" section is purely placeholder, while the rest of the quick start sections are in "draft: true" status and won't appear in prod. 

Addressing point 2: fixed. 

Re: v2 branch, sounds good. This will be my first base branch change so if anything fks up, my apologies!
